src/wallet/wallet.cpp

```
@@ -3161,6 +3161,8 @@ std::shared_ptr<CWallet> CWallet::Create(WalletContext& context, const std::stri
     }
 
     if (args.IsArgSet("-paytxfee")) {
+        warnings.push_back(_("-paytxfee is deprecated and will be fully removed in v31.0."));
+
```

In "wallet,rpc: paytxfee and settxfee is deprecated and hidden from help" ee080ce689bfc8b74438a02e1c62bbe644893ffa

Shouldn't this be a failure instead of a warning since an error is being thrown in the RPC? I think we should be consistent and fail as well. 

The user should explicitly provide something like `deprecatedstartupoption=paytxfee`. Do we support that?

---

I don't think we have something like `deprecatedstartupoption`.
I did it with a warn trying to follow the way it's done with Legacy wallets.
 https://github.com/bitcoin/bitcoin/blob/79bbb381a1fd13ad456fde736b3c195a791d4e58/src/wallet/wallet.cpp#L484-L487